spec/models/pro_account_spec.rb

```diff
@@ -74,6 +74,12 @@
         with(pro_account.user.email)
     end
 
+    it 'sets Stripe customer source' do
+      pro_account.source = mock_source = double(:source)
+      pro_account.update_stripe_customer
+      expect(mock_customer).to have_received(:source=).with(mock_source)
```

Comment:
Better to test the expected result like `expect(mock_customer.source).to eq(mock_source)` rather than the implementation details?

Reply:
Not dealing with `Stripe::Customer` objects here just mock null objects so your suggestion doesn't work without changing the spec setup to enable `StripeMock` and setup up everything as Stripe classes - this may well work now I have recently identified the issue I was having with `StripeMock` - would need to merge Gemfile change first

Reply:
As mentioned in https://github.com/mysociety/alaveteli/pull/5356#discussion_r320970653 this will be address in the other PR